tools/spectral/ipa/rulesets/functions/eachCustomMethodMustBeGetOrPost.js

```diff
@@ -0,0 +1,26 @@
+import { isCustomMethod } from './utils/resourceEvaluation.js';
+
+const ERROR_MESSAGE = 'The HTTP method for custom methods must be GET or POST.';
+const ERROR_RESULT = [{ message: ERROR_MESSAGE }];
+const VALID_METHODS = ['get', 'post'];
+
+export default (input, opts, { path }) => {
+  // Extract the path key (e.g., '/a/{exampleId}:method') from the JSONPath.
+  let pathKey = path[1];
+
+  if (!isCustomMethod(pathKey)) return;
+
+  const httpMethods = Object.keys(input);
+
+  // Check for invalid methods
+  if (httpMethods.some((method) => !VALID_METHODS.includes(method))) {
+    return ERROR_RESULT;
+  }
```

Comment:
Will this return error if the path has an exception extension? 

Suggestion: use `$.path[get,put,post,delete,options,head,patch,trace]` to ignore any other objects than the methods, kinda like we do here: https://github.com/mongodb/openapi/blob/98f5b237e447fc07b191833195a32fc310a4daa4/tools/spectral/.spectral.yaml#L9

Reply:
Yes, it would return error with an exception extension. Good catch!

Reply:
`"#PathItem[get,put,post,delete,options,head,patch,trace]" ` gives the object inside HTTP method. I will exclude `x-xgen-IPA-exception`